OTPKit/Features/TripPlanner/TripPlannerSheetView.swift

```diff
@@ -15,41 +15,130 @@ public struct TripPlannerSheetView: View {
+                                        switch leg.mode {
+                                        case "TRAM":
+                                            VStack {
+                                                HStack(spacing: 4) {
+                                                    Text(leg.route ?? "")
+                                                        .padding(.horizontal, 8)
+                                                        .padding(.vertical, 4)
+                                                        .background(Color.blue)
+                                                        .foregroundStyle(.foreground)
+                                                        .font(.caption)
+                                                        .clipShape(RoundedRectangle(cornerRadius: 4))
+
+                                                    Image(systemName: "tram")
+                                                        .foregroundStyle(.foreground)
+                                                }
+                                            }.frame(height: 40)
+
+                                        case "BUS":
+                                            HStack(spacing: 4) {
+                                                Text(leg.route ?? "")
+                                                    .padding(.horizontal, 8)
+                                                    .padding(.vertical, 4)
+                                                    .background(Color.green)
+                                                    .foregroundStyle(.foreground)
+                                                    .font(.caption)
+                                                    .clipShape(RoundedRectangle(cornerRadius: 4))
+
+                                                Image(systemName: "bus")
+                                                    .foregroundStyle(.foreground)
+                                            }.frame(height: 40)
+
+                                        case "WALK":
+                                            HStack(spacing: 4) {
+                                                Image(systemName: "figure.walk")
+                                                Text(formatTimeDuration(leg.duration))
+                                            }
+                                            .padding(.horizontal, 8)
+                                            .padding(.vertical, 4)
+                                            .background(Color.gray.opacity(0.2))
+                                            .foregroundStyle(.gray)
+                                            .clipShape(RoundedRectangle(cornerRadius: 8))
+                                            .frame(height: 40)
+
+                                        default:
+                                            Text("")
```

Comment:
What about other potential modes of transit? Might it be better to put something on the screen that indicates that `leg.mode` isn't something we know how to render?

Reply:
Right. That would be better. Thanks for the feedback!